cmd/run.go

```diff
@@ -137,249 +165,248 @@ func newRunCommand() *cobra.Command {
+			// - https://github.com/advisories/GHSA-qppj-fm5r-hxr3
+			// - https://github.com/advisories/GHSA-4374-p667-p6c8
+			var tlsOpts []func(*tls.Config)
+			disableHTTP2 := func(c *tls.Config) {
+				log.Infof("disabling http/2")
+				c.NextProtos = []string{"http/1.1"}
+			}
 
-			// Initialize metrics before starting the metrics server or using any counters
-			metrics.InitMetrics()
+			if !enableHTTP2 {
+				tlsOpts = append(tlsOpts, disableHTTP2)
+			}
 
-			// Health server will start in a go routine and run asynchronously
-			var hsErrCh chan error
-			var msErrCh chan error
-			if cfg.HealthPort > 0 {
-				log.Infof("Starting health probe server TCP port=%d", cfg.HealthPort)
-				hsErrCh = health.StartHealthServer(cfg.HealthPort)
+			webhookServer := webhook.NewServer(webhook.Options{
+				TLSOpts: tlsOpts,
+			})
+
+			// Metrics endpoint is enabled in 'config/default/kustomization.yaml'. The Metrics options configure the server.
+			// More info:
+			// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.19.0/pkg/metrics/server
+			// - https://book.kubebuilder.io/reference/metrics.html
+			metricsServerOptions := metricsserver.Options{
+				BindAddress:   metricsAddr,
+				SecureServing: secureMetrics,
+				// TODO(user): TLSOpts is used to allow configuring the TLS config used for the server. If certificates are
+				// not provided, self-signed certificates will be generated by default. This option is not recommended for
+				// production environments as self-signed certificates do not offer the same level of trust and security
+				// as certificates issued by a trusted Certificate Authority (CA). The primary risk is potentially allowing
+				// unauthorized access to sensitive metrics data. Consider replacing with CertDir, CertName, and KeyName
+				// to provide certificates, ensuring the server communicates using trusted and secure certificates.
+				TLSOpts: tlsOpts,
 			}
 
-			if cfg.MetricsPort > 0 {
-				log.Infof("Starting metrics server on TCP port=%d", cfg.MetricsPort)
-				msErrCh = metrics.StartMetricsServer(cfg.MetricsPort)
+			if secureMetrics {
+				// FilterProvider is used to protect the metrics endpoint with authn/authz.
+				// These configurations ensure that only authorized users and service accounts
+				// can access the metrics endpoint. The RBAC are configured in 'config/rbac/kustomization.yaml'. More info:
+				// https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.19.0/pkg/metrics/filters#WithAuthenticationAndAuthorization
+				metricsServerOptions.FilterProvider = filters.WithAuthenticationAndAuthorization
 			}
 
-			if warmUpCache {
-				err := warmupImageCache(cfg)
-				if err != nil {
-					log.Errorf("Error warming up cache: %v", err)
-					return err
-				}
+			mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+				Scheme:                  scheme,
+				Metrics:                 metricsServerOptions,
+				WebhookServer:           webhookServer,
```

Comment:
Will the manager start the webhook server? Is there a way to tell the manager to skip webhook server, at least for now, until we come to implement webhook support as a sepate feature?